### DISCUSSION TITLE
Progressive Performance Degradation and High CPU Usage with GitHub Copilot

### DISCUSSION THREAD

**Discussion by ovojinovic** [community/community#118893] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Dear GitHub Copilot team,

A few days ago, I started encountering significant and progressively worsening performance issues while using GitHub Copilot, primarily with .ipynb files. The issues manifest as follows:

1. **Lag and High CPU Usage:** There is a delay of up to 30 seconds before the execution of cells in .ipynb files or after initiation of Intellisense with CTRL+Space. During this lag, VSCode's CPU usage spikes to around 30%. Once the execution commences, the time taken for execution appears normal and the CPU usage drops back to regular levels. Additionally, there is a substantial delay in the appearance of Copilot suggestions, often taking between 30 seconds to a minute after I stop typing.

2. **Progressive Performance Degradation:** Over time, the performance issues intensify. Eventually, Visual Studio Code becomes entirely unresponsive. When I pause using VSCode for a couple of minutes, performance restores to initial.

3. **Slow Response in GitHub Copilot Chat:** Initially, the speed of answer generation in the GitHub Copilot chat is acceptable. However, it progressively slows down during the generation of a long answer, to one word per 1.5 seconds.

I have attempted to disable all other extensions, but the issue persists exclusively when GitHub Copilot is enabled, leading me to believe that the issue is specifically related to GitHub Copilot.

Here is some information about my setup:

Operating System: Microsoft Windows 10 Home Single Language build 19045.4170
Visual Studio Code Version: 1.88.1
GitHub Copilot Version: 1.180.0

Thank you for your time and assistance.

**Comment by Ilyes512** [comment]
I get the same thing... I was not sure what was causing this lag (on both my devices). I started getting to be to much... start turning off extension 1 by 1 until I turned off copilot... and voila, with the rest of my disabled extensions enabled again vscode was running smooth again.

This was really, really annoying and started to get in my way of getting my work done.

**Comment by notlousybook** [comment]
yeah same

**Comment by MinhTD18** [comment]
Did the copilot team have a solution for this? The Copilot extension for VS Code takes 80% of CPU usage on a universal Mac. My MacBook reach 61 degree Celcius after turn on the extension in 5 minutes

**Comment by Root-source-ai** [comment]
Exactly the same, I'm at 130% at code render when the copilot extension is on, it also says 715ms on the Copilot extension.

**Comment by JealousGx** [comment]
Any solution to this? I've been facing the same issue lately. My entire macbook becomes unresponsive because of this. I have to force quit and then re-open the vscode every 2 minutes due to copilot. It only occurs when I open github copilot chat. If I dont open it, it's working fine.

Thinking of just switching the editor since there's no response from the team on this issue reported year ago.
